src/core/net/udp6.cpp

```diff
@@ -533,16 +548,31 @@ void Udp::HandlePayload(Message &aMessage, MessageInfo &aMessageInfo)
     return;
 }
 
-bool Udp::IsMlePort(uint16_t aPort) const
+#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
+bool Udp::ShouldUsePlatformUdp(uint16_t aPort) const
 {
-    bool isMlePort = (aPort == Mle::kUdpPort);
-
+    return (aPort != Mle::kUdpPort && aPort != Tmf::kUdpPort
 #if OPENTHREAD_FTD
-    isMlePort = isMlePort || (aPort == Get<MeshCoP::JoinerRouter>().GetJoinerUdpPort());
+            && aPort != Get<MeshCoP::JoinerRouter>().GetJoinerUdpPort()
+#endif
+    );
+}
+
+bool Udp::ShouldUsePlatformUdp(const Udp::SocketHandle &aSocket) const
+{
+    uint16_t port       = aSocket.mSockName.mPort;
+    bool     usePlatUdp = false;
+
+    VerifyOrExit(!ShouldUsePlatformUdp(port), usePlatUdp = true);
+
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    usePlatUdp = IsBackboneSocket(aSocket);
```

Comment:
nit: what about something like (similar to `ShouldUsePlatformUdp(uint16_t aPort)` above):
```c++
    return (ShouldUsePlatformUdp(port)
#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
            || IsBackboneSocket(aSocket)
#endif
    );
```

Reply:
Yes. 
I changed to the current code to be more readable. But it seems actually less readable.

Reply:
updated